source/math/FloatBox2D.ooc

```diff
@@ -111,6 +111,20 @@ FloatBox2D: cover {
 				result add(i)
 		result
 	}
+	separatedDistance: func (point: FloatPoint2D) -> FloatPoint2D {
+		distanceX := Float absolute(point x - this center x) - this width / 2.f
+		distanceY := Float absolute(point y - this center y) - this height / 2.f
+		FloatPoint2D new(distanceX, distanceY) maximum(FloatPoint2D new(0.f, 0.f))
```

Comment:
However, you can do `(- this center + point) absolute`. Doesn't look very good, but that's kind of the point.